src/clang.rs

```diff
@@ -157,17 +155,18 @@ impl Cursor {
 
     /// Is the referent a top level construct?
     pub fn is_toplevel(&self) -> bool {
-        let mut semantic_parent = self.semantic_parent();
+        let mut semantic_parent = self.fallible_semantic_parent();
 
-        while semantic_parent.kind() == CXCursor_Namespace ||
-              semantic_parent.kind() == CXCursor_NamespaceAlias ||
-              semantic_parent.kind() == CXCursor_NamespaceRef {
-            semantic_parent = semantic_parent.semantic_parent();
+        while semantic_parent.is_some() &&
+              (semantic_parent.unwrap().kind() == CXCursor_Namespace ||
+               semantic_parent.unwrap().kind() == CXCursor_NamespaceAlias ||
+               semantic_parent.unwrap().kind() == CXCursor_NamespaceRef) {
+            semantic_parent = semantic_parent.unwrap().fallible_semantic_parent();
         }
 
         let tu = self.translation_unit();
-        // Yes, the second can happen with, e.g., macro definitions.
-        semantic_parent == tu || semantic_parent == tu.semantic_parent()
+        // Yes, this can happen with, e.g., macro definitions.
+        semantic_parent == tu.fallible_semantic_parent()
```

Comment:
At this point, `semantic_parent.unwrap()` is None. Therefore true can only be returned if tu is None or `tu.fallible_semantic_parent()` is `None`. `tu` will never be `None` because it is a `Cursor` type, so the first check is now unnecessary.